data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
-            defaults to "SMILES"
+
+def has_yaml_file_smiles_column(yaml_file: Union[str, Path]) -> bool:
+    """Returns True if the yaml file has a SMILES column.
+
+    Those columns are in either the targets or identifiers keys
+    (which are both lists). In there, we have dicts and it is a SMILES
+    if the key type is SMILES.
     """
+    with open(yaml_file, "r") as f:
+        meta = yaml.safe_load(f)
+
+    if "targets" in meta:
+        for target in meta["targets"]:
+            if target["type"] == "SMILES":
+                return True
+    if "identifiers" in meta:
+        for identifier in meta["identifiers"]:
+            if identifier["type"] == "SMILES":
+                return True
+
+    return False
+
+
+def is_in_scaffold_split_list(yaml_file: Union[str, Path]) -> bool:
+    """Returns True if the yaml file is in the to_scaffold_split list."""
+    with open(yaml_file, "r") as f:
+        meta = yaml.safe_load(f)
+
+    return meta["name"] in to_scaffold_split
 
-    for path in csv_paths:
-        read_dataset = pd.read_csv(path)
-
-    if check:
-        train_smiles = set(train_test_df.query("split == 'train'")[repr_col].to_list())
-
-    for path in csv_paths:
-        read_dataset = pd.read_csv(path)
-        if repr_col in read_dataset.columns:
-            try:
-                read_dataset = read_dataset.drop("split", axis=1)
-                message = f"Split column found in {path}."
-                if override:
-                    message += " Overriding..."
-                print(message)
-            except KeyError:
-                print(f"No split column in {path}")
-
-            merged_data = pd.merge(read_dataset, train_test_df, on=repr_col, how="left")
-            # merged_data = merged_data.dropna()
-            if override:
-                merged_data.to_csv(path, index=False)
-            else:
-                # rename the old data_clean.csv file to data_clean_old.csv
-                os.rename(path, path.replace(".csv", "_old.csv"))
-                # write the new data_clean.csv file
-                merged_data.to_csv(path, index=False)
 
-            if len(merged_data.query("split == 'train'")) == 0:
-                raise ValueError("Split failed, no train data")
-            if len(merged_data.query("split == 'test'")) == 0:
-                raise ValueError("Split failed, no test data")
-            if check:
-                test_split_smiles = set(
-                    merged_data.query("split == 'test'")[repr_col].to_list()
-                )
-                if len(train_smiles.intersection(test_split_smiles)) > 0:
-                    raise ValueError("Split failed, train and test overlap")
+def get_yaml_files(data_dir: Union[str, Path]) -> List[str]:
+    """Returns all yaml files in the data_dir directory."""
+    return glob(os.path.join(data_dir, "**", "*.yaml"), recursive=True)
 
 
-TRACKED_DATASETS = []
+def run_transform(file):
```

Comment:
I wouldn't run that here, this should be done beforehand.
I'm afraid if this gets run here we over ride a file, get a file we don't wanted in the first place, or this breaks because of some special dirs that have a different setup, e.g., bio ner, mol translation.

Reply:
This is now only an option.

Reply:
Optional now, and defaults to false